src/lib/actions/helpers/decorators/__tests__/decorateHtmlCode.test.js

```diff
@@ -0,0 +1,195 @@
+  });
+
+  it('does not replace data element tokens for an embedded html action', function() {
+    var settings = {
+      language: 'html',
+      source: '<div>%productname%</div>'
+    };
+
+    var decorateHtmlCode = decorateHtmlCodeInjector();
+    var decoratedResult = decorateHtmlCode(
+      {
+        settings: settings,
+        event: {},
+        relatedElement: {}
+      },
+      settings.source
+    );
+
+    expect(decoratedResult.code).toBe('<div>%productname%</div>');
+    expect(mockTurbine.replaceTokens).not.toHaveBeenCalled();
+  });
+
+  it('does replace data element tokens for an html action loaded from a file', function() {
+    var settings = {
+      language: 'html',
+      source: 'url1',
+      isExternal: true
+    };
+
+    var decorateHtmlCode = decorateHtmlCodeInjector();
+    var decoratedResult = decorateHtmlCode(
+      {
+        settings: settings,
+        event: {},
+        relatedElement: {}
+      },
+      '<div>%productname%</div>'
+    );
+
+    expect(decoratedResult.code).toBe('<div>replaced - productname</div>');
+  });
+
+  it(
+    'returns a resolved promise on the promise key when HTML code ' +
+      'does not contain callbacks',
+    function() {
+      var settings = {
+        language: 'html',
+        source: '<script>console.log("logging")</script>'
+      };
+
+      var p = Promise.resolve();
+      var decorateHtmlCode = decorateHtmlCodeInjector({
+        '@adobe/reactor-promise': {
+          resolve: function() {
+            return p;
+          }
+        }
+      });
+
+      var decoratedResult = decorateHtmlCode(
+        {
+          settings: settings
+        },
+        settings.source
+      );
+
+      expect(decoratedResult.promise).toBe(p);
+    }
+  );
+
+  it(
+    'returns a promise that will be resolved when HTML code contains callbacks' +
+      'and _satellite._onCustomCodeSuccess is called',
+    function(done) {
+      var settings = {
+        language: 'html',
+        source:
+          '<script>_satellite._onCustomCodeSuccess(${reactorCallbackId})</script>'
+      };
+
+      var decorateHtmlCode = decorateHtmlCodeInjector();
+
+      decorateHtmlCode(
+        {
+          settings: settings
+        },
+        settings.source
+      ).promise.then(function() {
+        done();
+      });
+
+      window._satellite._onCustomCodeSuccess(0);
+    }
+  );
+
+  it(
+    'returns a promise that will be rejected when HTML code contains callbacks' +
+      'and _satellite._onCustomCodeFailure is called',
+    function(done) {
+      var settings = {
+        language: 'html',
+        source:
+          '<script>_satellite._onCustomCodeFailure(${reactorCallbackId})</script>'
+      };
+
+      var decorateHtmlCode = decorateHtmlCodeInjector();
+
+      decorateHtmlCode(
+        {
+          settings: settings
+        },
+        settings.source
+      ).promise.catch(function() {
+        done();
+      });
+
+      window._satellite._onCustomCodeFailure(0);
```

Comment:
Same here. In isolation, this test doesn't really assert that the promise is failing as a result of the `window._satellite._onCustomCodeFailure(0);` call (it could have been rejected previously for other reasons, for example).